mpqp/execution/providers/ibm.py

```diff
@@ -420,84 +487,59 @@ def submit_remote_ibm(job: Job) -> tuple[str, "RuntimeJobV2"]:
-                " simulators and devices. Please use a local simulator instead."
-            )
     check_job_compatibility(job)
 
     qiskit_circ = job.circuit.to_other_language(Language.QISKIT)
     if TYPE_CHECKING:
         assert isinstance(qiskit_circ, QuantumCircuit)
 
     qiskit_circ = qiskit_circ.reverse_bits()
+
     service = get_QiskitRuntimeService()
+
     assert isinstance(job.device, IBMDevice)
     backend = get_backend(job.device)
     session = Session(service=service, backend=backend)
-    qiskit_circ = transpile(qiskit_circ, backend)
+
+    pm = generate_preset_pass_manager(backend=backend, optimization_level=1)
+    qiskit_circ = pm.run(qiskit_circ)
 
     if job.job_type == JobType.OBSERVABLE:
-        tot_size = qiskit_circ.num_qubits
         if TYPE_CHECKING:
             assert isinstance(meas, ExpectationMeasure)
-        estimator = Runtime_Estimator(session=session)
+        estimator = Runtime_Estimator(mode=session)
         qiskit_observable = meas.observable.to_other_language(Language.QISKIT)
         if TYPE_CHECKING:
             assert isinstance(qiskit_observable, SparsePauliOp)
 
-        # Fills the Pauli strings with identities to make the observable size
-        # match the circuit size
-        qiskit_observable = SparsePauliOp(
-            [
-                # for some reason, the type checker gets the type of
-                # pauli.paulis[0] wrong, and as such the wrong tensor is
-                # inferred. Because of this, the type inside it are not guessed
-                # properly either, forcing us to "ignore" this problem.
-                pauli.paulis[0].tensor(
-                    PauliList(  # pyright: ignore[reportArgumentType]
-                        Pauli("I" * (tot_size - meas.observable.nb_qubits))
-                    )
-                )
-                for pauli in qiskit_observable
-            ],
-            coeffs=qiskit_observable.coeffs,
-        )
+        qiskit_observable = qiskit_observable.apply_layout(qiskit_circ.layout)
 
-        # FIXME: when we precise the target precision like this, it does not give the right number of shots at the end.
-        #  https://github.com/Qiskit/qiskit-ibm-runtime/blob/ed71c5bf8d4fa23c26a0a26c6d45373263e5ecde/qiskit_ibm_runtime/qiskit/primitives/backend_estimator_v2.py#L154
-        #  Tried once with shots=1234, but got shots=1280 with the real experiment, looks like the decimal part of
-        #  precision is truncated. The problem is on the IBM side, an issue has been published :
-        #  https://github.com/Qiskit/qiskit-ibm-runtime/issues/1749
-        precision = 1 / np.sqrt(meas.shots)
-        ibm_job = estimator.run([(qiskit_circ, qiskit_observable)], precision=precision)
+        # We have to disable all the twirling options and set manually the number of circuits and shots per circuits
+        twirling = getattr(estimator.options, "twirling", None)
```

Comment:
the `getattr` and `setattr` are because the types of qiskit are somewhat messed up ?